CoreBluetoothMock/CBMPeripheralSpecDelegate.swift

```diff
@@ -251,155 +251,113 @@ public protocol CBMPeripheralSpecDelegate {
-                    data: Data) {
-        peripheral(p, didReceiveWriteCommandFor: characteristic as CBMCharacteristic, data: data)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveWriteCommandFor characteristic: CBMCharacteristicMock, data: Data) {
+        peripheral(p, didReceiveWriteCommandFor: characteristic, data: data)
     }
     
-    func peripheral(_ peripheral: CBMPeripheralSpec,
-                    didReceiveWriteRequestFor descriptor: CBMDescriptor,
-                    data: Data)
-        -> Result<Void, Error> {
-            return .failure(CBATTError(.writeNotPermitted))
+    func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveWriteRequestFor descriptor: CBMDescriptor, data: Data) -> Result<Void, Error> {
+            
+        .failure(CBATTError(.writeNotPermitted))
     }
     
-    func peripheral(_ p: CBMPeripheralSpec,
-                    didReceiveWriteRequestFor descriptor: CBMDescriptorMock,
-                    data: Data)
-        -> Result<Void, Error> {
-            return peripheral(p, didReceiveWriteRequestFor: descriptor as CBMDescriptor, data: data)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveWriteRequestFor descriptor: CBMDescriptorMock, data: Data) -> Result<Void, Error> {
+            
+        peripheral(p, didReceiveWriteRequestFor: descriptor, data: data)
     }
     
-    func peripheral(_ peripheral: CBMPeripheralSpec,
-                    didReceiveSetNotifyRequest enabled: Bool,
-                    for characteristic: CBMCharacteristic)
-        -> Result<Void, Error> {
-            if !characteristic.properties
-                .isDisjoint(with: [
-                    .notify,
-                    .indicate,
-                    .notifyEncryptionRequired,
-                    .indicateEncryptionRequired]) {
-                return .success(())
-            } else {
-                return .failure(CBMError(.invalidHandle))
-            }
+    func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveSetNotifyRequest enabled: Bool, for characteristic: CBMCharacteristic) -> Result<Void, Error> {
+            
+        if !characteristic.properties.isDisjoint(with: [.notify, .indicate, .notifyEncryptionRequired, .indicateEncryptionRequired]) {
+            return .success(())
+        } else {
+            return .failure(CBMError(.invalidHandle))
+        }
     }
     
-    func peripheral(_ p: CBMPeripheralSpec,
-                    didReceiveSetNotifyRequest enabled: Bool,
-                    for characteristic: CBMCharacteristicMock)
-        -> Result<Void, Error> {
-            return peripheral(p, didReceiveSetNotifyRequest: enabled, for: characteristic as CBMCharacteristic)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveSetNotifyRequest enabled: Bool, for characteristic: CBMCharacteristicMock) -> Result<Void, Error> {
+            
+        peripheral(p, didReceiveSetNotifyRequest: enabled, for: characteristic)
```

Comment:
And again here. This will cause stack overflow due to recursion instead of intended support for calling deprecated methods.